pkg/foundation/grpcutil/websocket.go

```diff
@@ -109,9 +113,18 @@ func (p *webSocketProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // and stop writing any data to the WebSocket connection. This will
 // automatically halt all the "pipeline nodes" after the underlying response.
 func (p *webSocketProxy) proxy(w http.ResponseWriter, r *http.Request) {
-	ctx, cancelFn := context.WithCancel(r.Context())
-	defer cancelFn()
+	ctx, cancelCtx := context.WithCancel(r.Context())
+	defer cancelCtx()
 	r = r.WithContext(ctx)
+	go func() {
+		// we're using ChanOut.Recv() so that the goroutine doesn't wait on
+		// the proxy to be done even when the request is done
+		_, _, err := cchan.ChanOut[struct{}](p.done).Recv(ctx)
+		if err != nil {
+			p.logger.Debug(ctx).Msgf("request context returned an error: %v", err)
+		}
```

Comment:
I think we should output the log regardless of the error and attach the error to the log message (if the error is `nil` it won't be attached). This way we will see that the connection was closed regardless of what triggered it and if there was an error we will also see it.
```suggestion
		p.logger.Debug(ctx).Err(err).Msgf("websocket connection closed")
```

To be precise, if `err` is `nil` it means the connection _will be closed_ (next line closes it), otherwise the connection was probably already closed. But I think we don't need to adjust our log message because of that, we will see this based on the attached error.